database/src/main/postgres/runs/V1.9.7__create_partitioning.sql

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2021 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+CREATE OR REPLACE FUNCTION runs.create_partitioning(
+    IN  i_partitioning          JSONB,
+    IN  i_by_user               TEXT,
+    IN  i_parent_partitioning   JSONB = NULL,
+    OUT status                  INTEGER,
+    OUT status_text             TEXT,
+    OUT id_partitioning         BIGINT
+) RETURNS record AS
+$$
+    -------------------------------------------------------------------------------
+--
+-- Function: runs.create_partitioning(3)
+--      Creates a partitioning entry
+--
+-- Parameters:
+--      i_partitioning          - partitioning to create or which existence to check
+--      i_by_user               - user behind the change
+--      i_parent_partitioning   - parent partitioning of the provided partitioning, optional
+--
+-- Returns:
+--      status              - Status code
+--      status_text         - Status text
+--      id_partitioning     - id of the partitioning
+--
+-- Status codes:
+--      11                  - Partitioning created
+--      12                  - Partitioning parent registered
+--      31                  - Partitioning already present
+--
+-------------------------------------------------------------------------------
+DECLARE
+    _fk_parent_partitioning BIGINT := NULL;
+    _create_partitioning    BOOLEAN;
```

Comment:
A couple of suggestions to make the code easier to follow, IMHO (will be marked with 👓 )
👓
```suggestion
    _partitioning_created   BOOLEAN;
```